src/lotus/client.rs

```diff
@@ -0,0 +1,148 @@
+use crate::message::{
+    CIDMap, MpoolPushMessage, MpoolPushMessageInner, MpoolPushMessageResponse, ReadStateResponse,
+    StateWaitMsgResponse, WalletKeyType, WalletListResponse,
+};
+use crate::{JsonRpcClient, LotusClientApi};
+use anyhow::Result;
+use async_trait::async_trait;
+use cid::Cid;
+use fvm_shared::address::Address;
+use fvm_shared::econ::TokenAmount;
+use num_traits::cast::ToPrimitive;
+use serde::de::DeserializeOwned;
+use serde_json::json;
+use std::fmt::Debug;
+use std::str::FromStr;
+
+// RPC methods
+mod methods {
+    pub const MPOOL_PUSH_MESSAGE: &str = "Filecoin.MpoolPushMessage";
+    pub const STATE_WAIT_MSG: &str = "Filecoin.StateWaitMsg";
+    pub const WALLET_NEW: &str = "Filecoin.WalletNew";
+    pub const WALLET_LIST: &str = "Filecoin.WalletList";
+    pub const WALLET_DEFAULT_ADDRESS: &str = "Filecoin.WalletDefaultAddress";
+    pub const STATE_READ_STATE: &str = "Filecoin.StateReadState";
+}
+
+/// The struct implementation for Lotus Client API. It allows for multiple different trait
+/// extension.
+pub struct LotusJsonRPCClient<T: JsonRpcClient> {
+    inner: T,
```

Comment:
`inner` is still a not very descriptive name. I suggest using `json_rpc_client` or just `client`.